### DISCUSSION TITLE
Local data

### DISCUSSION THREAD

**Discussion by pape-az** [aleksey-saenko/MusicRecognizer#64] [Q&A]
Can we have a local database without depending on an API ? Should we manually put music there ?

**Comment by aleksey-saenko** [comment]
Do you mean offline music recognition, using its own database of fingerprints without relying on a third-party service?

**Reply by pape-az** [reply]
Exactly

Also have a dashboard to see how many times a sound was played

**Comment by aleksey-saenko** [comment]
Music recognition is difficult to do offline, as databases of audio fingerprints are huge and require continuous updates. In the case of mobile devices, the client-server interaction seems quite justified here. Therefore, Audile is only a client for services that perform recognition.

Having your own remote database is also not feasible, as creating and maintaining such a database is a huge job. That is why all popular recognition services are not available without a fee. There are open services supported by the community, for example [AcoustID](https://acoustid.org). But this service works only for original files, not for noisy samples taken from the microphone.

The closest option to your request I see is using the [AudioTag](https://audiotag.info/) service, which is almost free and allows users to contribute to the remote database. But when I last tried AudioTag, it seemed to me much worse in terms of recognition than the already used AudD and ACRcloud. The database is much smaller, the API is inconvenient, and the track's metadata is scant.

**Reply by pape-az** [reply]
AcoustID and AudioTag are already integrated ?

**Reply by aleksey-saenko** [reply]
No, I did not integrate them into Audile. AcoustID is not designed for recognition from a microphone, and AudioTag was not added for the reasons stated above. However, maybe someday AudioTag support will be implemented. 
At the moment there is only support for AudD and ACRCloud. They're much better.

**Reply by pape-az** [reply]
Tremendous
